crates/volta-core/src/tool/package/fetch.rs

```diff
@@ -80,9 +77,95 @@ fn load_cached_distro(file: &Path, shasum_file: &Path) -> Option<Box<dyn Archive
     Tarball::load(distro).ok()
 }
 
-fn fetch_remote_distro(spec: tool::Spec, url: &str, path: &Path) -> Fallible<Box<dyn Archive>> {
-    debug!("Downloading {} from {}, to {}", &spec, &url, path.display());
-    Tarball::fetch(url, path).with_context(download_tool_error(spec, url.to_string()))
+fn fetch_remote_distro(
+    path: &Path,
+    name: &str,
+    details: &PackageDetails,
+    session: &mut Session,
+) -> Fallible<Box<dyn Archive>> {
+    ensure_containing_dir_exists(&path).with_context(|_| ErrorDetails::ContainingDirError {
+        path: path.to_path_buf(),
+    })?;
+
+    let dir = path.parent().unwrap();
```

Comment:
Can we add a comment here noting that the `parent()` will _always_ be `Some` because `ensure_containing_dir_exists` will error if there isn't a containing dir? We generally try to avoid using `unwrap` since it can potentially panic, but in this case, we know for sure that if we got here then it won't panic.